test/integration/client-side-encryption/driver.test.ts

```diff
@@ -0,0 +1,297 @@
+       *     clientId: string;
+       *     clientSecret: string;
+       *   };
+       *   gcp: {
+       *     email: string;
+       *     privateKey: string;
+       *   };
+       *   local: {
+       *     // EJSON handle converting this, its actually the canonical -> { $binary: { base64: string; subType: string } }
+       *     // **NOTE**: The dollar sign has to be escaped when using this as an ENV variable
+       *     key: Binary;
+       *   }
+       * }
+       * ```
+       */
+      expect(() => EJSON.parse(CSFLE_KMS_PROVIDERS)).to.not.throw(SyntaxError);
+    } else {
+      this.skip();
+    }
+  });
+
+  describe('Collection', metadata, function () {
+    describe('#bulkWrite()', metadata, function () {
+      context('when encryption errors', function () {
+        let client: MongoClient;
+
+        beforeEach(function () {
+          client = this.configuration.newClient(
+            {},
+            {
+              autoEncryption: {
+                keyVaultNamespace: 'test.keyvault',
+                kmsProviders: {
+                  local: {
+                    key: 'A'.repeat(128)
+                  }
+                },
+                encryptedFieldsMap: {
+                  'test.coll': {
+                    fields: [
+                      {
+                        path: 'ssn',
+                        keyId: new UUID('23f786b4-1d39-4c36-ae88-70a663321ec9').toBinary(),
+                        bsonType: 'string'
+                      }
+                    ]
+                  }
+                }
+              }
+            }
+          );
+        });
+
+        afterEach(async function () {
+          await client.close();
+        });
+
+        it('bubbles up the error', metadata, async function () {
+          try {
+            await client
+              .db('test')
+              .collection('coll')
+              // @ts-expect-error: Incorrectly formatted bulkWrite to test error case
+              .bulkWrite([{ insertOne: { ssn: 'foo' } }]);
+            expect.fail('expected error to be thrown');
+          } catch (error) {
+            expect(error.name).to.equal('MongoBulkWriteError');
+          }
+        });
+      });
+    });
+  });
+
+  describe('BSON Options', function () {
+    let client: MongoClient;
+    let encryptedClient: MongoClient;
+
+    beforeEach(function () {
+      client = this.configuration.newClient();
+
+      const noop = () => null;
+      function encryptSchema(keyId, bsonType) {
+        return {
+          encrypt: {
+            bsonType,
+            algorithm: 'AEAD_AES_256_CBC_HMAC_SHA_512-Random',
+            keyId: [keyId]
+          }
+        };
+      }
+
+      let encryption: ClientEncryption;
+      let dataDb: Db;
+      let keyVaultDb: Db;
+
+      const mongodbClientEncryption = this.configuration.mongodbClientEncryption;
+      const kmsProviders = this.configuration.kmsProviders(crypto.randomBytes(96));
+      return client
```

Comment:
just wondering why we aren't using async/await here instead

Reply:
This was a drive by TS convert, but don't mind if I do!